### PR TITLE
add instructions about how to define output file when running scrapy …

### PR DESCRIPTION
**Pull request by MagdalenaDeschner**
…from script instead of cmd

The following stackoverflow questions show, that it is unclear how to define the feed format and name within a script. Equivalent to "-o" flag when starting crawler process within cmd.
https://stackoverflow.com/questions/43468275/running-scrapy-from-a-script-with-file-output/43468852
https://stackoverflow.com/questions/23574636/scrapy-from-script-output-in-json

### CODE REVIEW

File: docs/topics/practices.rst [hunk 1]
```diff
@@ -35,12 +35,17 @@ Here's an example showing how to run a single spider with it.
         ...
 
     process = CrawlerProcess({
-        'USER_AGENT': 'Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)'
+        'USER_AGENT': 'Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)',
```
Comment (Gallaecio) [inline docs/topics/practices.rst]: What about removing `USER_AGENT` as part of these changes? I believe the point of `USER_AGENT` was make it obvious that the first parameter of `CrawlerProcess` is a dictionary of settings, which the new settings already accomplish.

File: docs/topics/practices.rst [hunk 2]
```diff
@@ -35,12 +35,17 @@ Here's an example showing how to run a single spider with it.
...
+        'FEED_FORMAT':'json',
+        'FEED_URI':'items.json'
     })
 
     process.crawl(MySpider)
     process.start() # the script will block here until the crawling is finished
 
+Define settings within dictionary in
+CrawlerProcess. FEED_FORMAT and FEED_URI are the equivalent to "-o items.json" when using the scrapy crawl shell command.
```
Comment (Gallaecio) [inline docs/topics/practices.rst]: > Define settings within dictionary in CrawlerProcess.

I think the example makes this clear, and the paragraph below does encourage reading the documentation of `CrawlerProcess`.

> FEED_FORMAT and FEED_URI are the equivalent to "-o items.json" when using the scrapy crawl shell command.

This information here feels out of the blue to me. If we really think that users will continue to have doubts despite the example featuring `FEED_FORMAT` and `FEED_URI`, then maybe we should add an entry to the FAQ instead, and really address the specific issue of “How to I use -o with CrawlProcess?”.

**Comment by MagdalenaDeschner** [comment]
@Gallaecio I totally agree with you, telling that you can define any setting within a dictionary as a parameter of CrawlerProcess should be enough. There shouldn't be an extra emphasis on any specific setting. Shall I make a new pull request or can you mark it as "needs work"?

File: docs/topics/practices.rst (hunk 1 above)
Comment (MagdalenaDeschner) [reply docs/topics/practices.rst]: maybe we can add CrawlerProcess(settings={}) so that it's totally clear.

File: docs/topics/practices.rst (hunk 2 above)
Comment (MagdalenaDeschner) [reply docs/topics/practices.rst]: I totally agree with you, telling that you can define any setting within a dictionary as a parameter of CrawlerProcess should be enough. There shouldn't be an extra emphasis on any specific setting. Shall I make a new pull request or can you mark it as "needs work"?

**Comment by Gallaecio** [comment]
> Shall I make a new pull request or can you mark it as "needs work"?

Please use the same pull request.

You can edit the title and prefix it with `[WIP]` to indicate that is currently _work in progress_, so that people don’t bother reviewing it until you remote the prefix from the title again.

File: docs/topics/practices.rst [hunk 3]
```diff
@@ -34,15 +34,16 @@ Here's an example showing how to run a single spider with it.
         # Your spider definition
         ...
 
-    process = CrawlerProcess({
-        'USER_AGENT': 'Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)'
+    process = CrawlerProcess(settings={
+        'FEED_FORMAT':'json',
+        'FEED_URI':'items.json'
```
Comment (kmike) [inline docs/topics/practices.rst]: a nitpick: could you please add whitespace after `:`, as per pep8?

Comment (MagdalenaDeschner) [reply docs/topics/practices.rst]: sure

**Comment by kmike** [comment]
Thanks @MagdalenaDeschner, and thanks @Gallaecio for a review!

**Pull request merged**